src/linux/routing/queueing/internal.hpp

```diff
@@ -118,7 +120,69 @@ Try<Netlink<struct rtnl_qdisc>> encodeDiscipline(
 }
 
 /////////////////////////////////////////////////
-// Helpers for internal APIs.
+// Helpers for {en}decoding queueing classes.
+/////////////////////////////////////////////////
+
+namespace cls {
+// Forward declaration. Each type of queueing class needs to implement
+// this function to encode its type specific configurations into the
+// libnl queueing class (rtnl_class).
+template <typename Config>
+Try<Nothing> encode(
+    const Netlink<struct rtnl_class>& cls,
+    const Config& config);
+
+
+// Forward declaration. Each type of queueing class needs to implement
+// this function to decode its type specific configurations from the
+// libnl queueing class (rtnl_class). Returns None if the libnl
+// queueing class does not match the specific queueing class type.
+template <typename Config>
+Result<Config> decode(const Netlink<struct rtnl_class>& cls);
+
+
+// Encodes a queueing class (in our representation) to a libnl
+// queueing class (rtnl_class). We use templating here so that it works
+// for any type of queueing class.
+template <typename Config>
+Try<Netlink<struct rtnl_class>> encode(
+    const Netlink<struct rtnl_link>& link,
+    const Class<Config>& config)
+{
+  struct rtnl_class* c = rtnl_class_alloc();
```

Comment:
here is that alloc I'm referring to that needs a corresponding put, but it looks like we make the caller deal with when to call the cleanup function